File: Codigo/Obligatorio/codigo/ArenaGestorFront/src/app/models/Concerts/ConcertUpdateConcertDto.ts
import { ConcertUpdateLocationDto } from "./ConcertUpdateLocationDto";
import { ConcertUpdateProtagonistDto } from "./ConcertUpdateProtagonistDto";

export class ConcertUpdateConcertDto {
    concertId: Number = 0;
    tourName: String = "";
    date: Date = new Date();
    ticketCount: Number = 0;
    price: Number = 0;
    protagonists: Array<ConcertUpdateProtagonistDto> = new Array();
    location: ConcertUpdateLocationDto = new ConcertUpdateLocationDto();
}

File: Codigo/Obligatorio/codigo/ArenaGestor/ArenaGestor.APIContracts/Concert/ConcertUpdateConcertDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ArenaGestor.APIContracts.Concert
{
    public class ConcertUpdateConcertDto
    {
        public int ConcertId { get; set; }

        [MaxLength(50)]
        [Required]
        public string TourName { get; set; }
        [Required]
        public DateTime Date { get; set; }
        [Required]
        public int TicketCount { get; set; }
        [Required]
        public double Price { get; set; }

        [Required]
        public List<ConcertUpdateProtagonistDto> Protagonists { get; set; }

        [Required]
        public ConcertUpdateLocationDto Location { get; set; }

    }
}
